src/managers/backup.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helpers for managing backups."""
+
+import logging
+
+import boto3
+from botocore import loaders, regions
+from botocore.exceptions import ClientError
+from mypy_boto3_s3.service_resource import Bucket
+
+from core.cluster import ClusterState
+
+logger = logging.getLogger(__name__)
+
+
+class BackupManager:
+    """Manager for all things backup-related."""
+
+    def __init__(self, state: ClusterState) -> None:
```

Comment:
**suggestion** It seems that we don't need the entire state for the backup manager, but just the s3-credentials. I would honestly nudge towards passing a smaller scoped input, such that we decouple more then two implementations. If you just use s3-credentials, we could actually take the very same class to a different charm (that has a different `ClusterState` though)

Reply:
That's a fair point, I narrowed the scope to just the S3 info in c67a0e7e9f12c2092637ac41402405a34b6964a5